include/endian.h

```diff
@@ -52,36 +52,36 @@
 /* Common byte swapping macros */
 
 #ifdef CONFIG_HAVE_BUILTIN_BSWAP16
-#  define __swap_uint16 __builtin_bswap16
+#  define __swap_uint16(n) ((uint16_t)__builtin_bswap16(n))
```

Comment:
This change is super strange. What is the warning here?

Reply:
Here, the return type of __builtin_bswap32 is 'unsigned int' not 'unsigned long'(uint32_t):
```
    Error: route/net_del_ramroute.c:94:9: error: format specifies type 'unsigned long' but the argument has type 'unsigned int' [-Werror,-Wformat]
            HTONL(match->target), HTONL(match->netmask));
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /github/workspace/sources/nuttx/include/debug.h:232:23: note: expanded from macro 'ninfo'
                          ^
    /github/workspace/sources/nuttx/include/debug.h:102:43: note: expanded from macro '_none'
        do { if (0) syslog(LOG_ERR, format, ##__VA_ARGS__); } while (0)
                                    ~~~~~~    ^~~~~~~~~~~
    /github/workspace/sources/nuttx/include/netinet/in.h:233:17: note: expanded from macro 'HTONL'
                    ^
    /github/workspace/sources/nuttx/include/endian.h:63:25: note: expanded from macro '__swap_uint32'
                            ^
    Error: route/net_del_ramroute.c:94:31: error: format specifies type 'unsigned long' but the argument has type 'unsigned int' [-Werror,-Wformat]
            HTONL(match->target), HTONL(match->netmask));
            ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
    /github/workspace/sources/nuttx/include/debug.h:232:23: note: expanded from macro 'ninfo'
                          ^
    /github/workspace/sources/nuttx/include/debug.h:102:43: note: expanded from macro '_none'
        do { if (0) syslog(LOG_ERR, format, ##__VA_ARGS__); } while (0)
                                    ~~~~~~    ^~~~~~~~~~~
    /github/workspace/sources/nuttx/include/netinet/in.h:233:17: note: expanded from macro 'HTONL'
                    ^
    /github/workspace/sources/nuttx/include/endian.h:63:25: note: expanded from macro '__swap_uint32'
```

Reply:
According to https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html `__builtin_bswap32` has the next prototype: `uint32_t __builtin_bswap32 (uint32_t x)`, so it is exactly `uint32_t` and not `unsigned int` or `unsigned long`. Maybe the correct way is to fix format message from
```
  ninfo("  target=%08lx netmask=%08lx\n",
        HTONL(match->target), HTONL(match->netmask));
```
to
```
  ninfo("  target=%08" PRIx32 " netmask=%08" PRIx32 "\n",
        HTONL(match->target), HTONL(match->netmask));
```
?

Reply:
Done.